dask/bag/core.py

```diff
@@ -1082,6 +1082,51 @@ def from_sequence(seq, partition_size=None, npartitions=None):
     return Bag(d, name, len(d))
 
 
+def from_castra(x, columns=None, index=False, npartitions=None):
+    """Load a dask Bag from a Castra.
+
+    Parameters
+    ----------
+    x : filename or Castra
+    columns: list or string, optional
+        The columns to load. Default is all columns.
+    index: bool, optional
+        If True, the index is included as the first element in each tuple.
+        Default is False.
+    npartitions: int, optional
+        The number of desired partitions. Defaults to number of partitions in
+        the Castra.
+    """
+    from castra import Castra
+    if not isinstance(x, Castra):
+        x = Castra(x)
+    path = x.path
+    if columns is None:
+        columns = x.columns
+    if npartitions is None:
+        npartitions = len(x.partitions)
+    parts = from_sequence(x.partitions, npartitions=npartitions)
+    func = lambda p: load_castra_partition(path, p, columns, index)
+    return parts.map_partitions(func).map_partitions(list)
```

Comment:
I've been trying to shift over to `reify` instead of `list` for this sort of thing.  Not sure what's best.